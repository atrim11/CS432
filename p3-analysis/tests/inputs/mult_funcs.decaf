// multiple functions with dup params
def int foo(int foo, int a, int b) {
    return foo(0, 0, 0);
}

def int bar(int a, int a, int b) {
    return 0;
}

def int main() {
    return 0;
}