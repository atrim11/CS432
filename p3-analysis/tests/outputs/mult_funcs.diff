--- outputs/mult_funcs.txt	2024-10-25 12:29:22.563935329 -0400
+++ expected/mult_funcs.txt	2024-10-25 12:16:00.420565807 -0400
@@ -1,3 +1,3 @@
-Invalid: Function call on line 3 has incorrect number of arguments
 Invalid call to non-function 'foo' on line 3
+Invalid number of function arguments on line 3
 Duplicate symbols named 'a' in scope started on line 6
