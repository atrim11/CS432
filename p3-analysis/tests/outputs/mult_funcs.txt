Invalid number of function arguments on line 3
Invalid call to non-function 'foo' on line 3
Duplicate symbols named 'a' in scope started on line 6
