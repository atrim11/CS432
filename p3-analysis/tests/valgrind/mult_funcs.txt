==288409== Memcheck, a memory error detector
==288409== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==288409== Using Valgrind-3.22.0 and LibVEX; rerun with -h for copyright info
==288409== Command: ../decaf inputs/mult_funcs.decaf
==288409== 
Invalid number of function arguments on line 3
Invalid call to non-function 'foo' on line 3
Duplicate symbols named 'a' in scope started on line 6
==288409== 
==288409== HEAP SUMMARY:
==288409==     in use at exit: 0 bytes in 0 blocks
==288409==   total heap usage: 2,009 allocs, 2,009 frees, 248,672 bytes allocated
==288409== 
==288409== All heap blocks were freed -- no leaks are possible
==288409== 
==288409== For lists of detected and suppressed errors, rerun with: -s
==288409== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
