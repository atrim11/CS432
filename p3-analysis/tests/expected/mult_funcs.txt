Invalid: Function call on line 3 has incorrect number of arguments
Invalid call to non-function 'foo' on line 3
Duplicate symbols named 'a' in scope started on line 6
